bitnami/ghost/templates/deployment.yaml

```diff
@@ -1,197 +1,189 @@
 {{- if include "ghost.host" . -}}
-apiVersion: {{ template "ghost.deployment.apiVersion" . }}
+apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
 kind: Deployment
 metadata:
-  name: {{ template "ghost.fullname" . }}
-  labels:
-    app: "{{ template "ghost.fullname" . }}"
-    chart: "{{ template "ghost.chart" . }}"
-    release: {{ .Release.Name | quote }}
-    heritage: {{ .Release.Service | quote }}
+  name: {{ include "ghost.fullname" . }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
 spec:
   selector:
-    matchLabels:
-      app: "{{ template "ghost.fullname" . }}"
-      release: {{ .Release.Name | quote }}
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
   replicas: 1
   template:
     metadata:
-      labels:
-        app: "{{ template "ghost.fullname" . }}"
-        chart: "{{ template "ghost.chart" . }}"
-        release: {{ .Release.Name | quote }}
+      labels: {{- include "common.labels.standard" . | nindent 8 }}
+    {{- if .Values.podAnnotations }}
+    annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.podAnnotations "context" $ ) | nindent 6 }}
+    {{- end }}
     spec:
-      {{- if .Values.securityContext.enabled }}
+    {{- include "ghost.imagePullSecrets" . | indent 6 }}
+    {{- if .Values.securityContext.enabled }}
       securityContext:
         fsGroup: {{ .Values.securityContext.fsGroup }}
         runAsUser: {{ .Values.securityContext.runAsUser }}
-      {{- else }}
+      {{- if .Values.initContainers }}
+      initContainers: {{- include "common.tplvalues.render" ( dict "value" .Values.initContainers "context" $ ) | nindent 8 }}
+      {{- end }}
+    {{- else }}
       initContainers:
-      - name: volume-permissions
-        image: {{ template "ghost.volumePermissions.image" . }}
-        imagePullPolicy: "{{ .Values.volumePermissions.image.pullPolicy }}"
-        command: ['sh', '-c', 'chmod -R g+rwX {{ .Values.persistence.path }}']
-        volumeMounts:
-        - mountPath: {{ .Values.persistence.path }}
-          name: ghost-data
+        - name: volume-permissions
+          image: {{ include "ghost.volumePermissions.image" . }}
+          imagePullPolicy: "{{ .Values.volumePermissions.image.pullPolicy }}"
+          command: ['sh', '-c', 'chmod -R g+rwX {{ .Values.persistence.path }}']
+          volumeMounts:
+            - mountPath: {{ .Values.persistence.path }}
+              name: ghost-data
+      {{- if .Values.initContainers }}
```

Comment:
I've thinking a lot about this recently. How problematic would it be to put initContainers outside the if so you don't have to repeat the .Values.initContainers section twice? The worst that can happen is to get an empty initContainers yaml section, which I don't think is an issue (not in our linters) and the template code would be simpler. Thoughts?

Reply:
yes, I think it could work fine and it would be the cleanest way, I did a try and an empty `initContainer` seems to not be a problem.

```yaml
    spec:
      {{- include "ghost.imagePullSecrets" . | indent 6 }}
      {{- if .Values.securityContext.enabled }}
      securityContext:
        fsGroup: {{ .Values.securityContext.fsGroup }}
        runAsUser: {{ .Values.securityContext.runAsUser }}
      {{- end }}
      initContainers:
      {{- if not .Values.securityContext.enabled }}
        - name: volume-permissions
          image: {{ include "ghost.volumePermissions.image" . }}
          imagePullPolicy: "{{ .Values.volumePermissions.image.pullPolicy }}"
          command: ['sh', '-c', 'chmod -R g+rwX {{ .Values.persistence.path }}']
          volumeMounts:
            - mountPath: {{ .Values.persistence.path }}
              name: ghost-data
      {{- end }}
      {{- if .Values.initContainers }}
        {{- include "common.tplvalues.render" ( dict "value" .Values.initContainers "context" $ ) | indent 8 }}
      {{- end }}
```

```bash
$ echo "$(helm template g   --set service.type=ClusterIP   --set securityContext.enabled=true   -s templates/deployment.yaml   bitnami/ghost)" | kubeval -
# PASS - ghost/templates/deployment.yaml contains a valid Deployment
```

Thanks!